### DISCUSSION TITLE
Open several artifacts like rosbags in Foxglove Studio at once

### DISCUSSION THREAD

**Discussion by matawed** [foxglove/community#834] [Q&A]
I'd like to be able to compare artifacts within Foxglove. Imagine that you run different software versions (e.g. one is a bugfix)  and each produces an artifact like a Rosbag. Each Rosbag then has the same signals and also maybe the same time base. Your task is then to make sure that you e.g. fix a problem or improve a previously not so good handled situation. 

Then I'd like to be able to check results from both runs in Foxglove and compare signals side by side or something similar. 

Several other tools like e.g. [PlotJuggler](https://github.com/facontidavide/PlotJuggler) make it possible to open several Rosbags and visualize their data at once.  You then get a user dialogue like 
![image](https://github.com/foxglove/community/assets/30142056/7279159a-5d20-4755-8746-019b0771597c)

as both artifacts contain same topics and are then no longer distinguishable by their name. After that you can then visualize data from both artifacts with selected prefix.

Currently, I don't know how to do that in Foxglove without prior Rosbag merge.

**Comment by jtbandes** [comment]
See also: https://github.com/orgs/foxglove/discussions/160

**Comment by matawed** [comment]
Thanks for referencing a similar discussion. Just to make sure that we have a common understanding.

I don't want to have this diff feature mentioned [here](https://github.com/orgs/foxglove/discussions/160#discussioncomment-4416547) or something else on top of current Foxglove Studio. It should "only" be able to load several bags and recognize if both have same topic names in it. If so, it should open a dialogue as mentioned in description of this PR. Nothing more and nothing less. 

Comparison of signals should then be done by users manually.

**Reply by defunctzombie** [reply]
FWIW we do allow opening multiple mcap files (via the file open dialog) now. We've not yet done this for ROS1 .bag files yet.
